next-docs/pages/templates/typescript/contracts.mdx

```diff
@@ -1 +1,453 @@
-# Contracts
+# Contracts (Vanilla and React)
+
+This section explains the contracts for the vanilla and react templates.
+The Three.JS template uses diffeerent contracts.
+
+The onchain components can be divided into two types of functionality:
+
+- **Data**, the part of the system that stores the information.
+- **Logic**, the part of the system that can be called to read or modify the data.
```

Comment:
I know you mean system in the general sense here, but i feel like it might be confusing with the "logic contracts" being called `System` as well. How about something in the direction of this:

```suggestion
- **Tables**, storing the data of the application.
- **Systems**, business logic that can be called to read or modify data in the tables.
```